src/Fbgemm.cc

```diff
@@ -255,6 +255,54 @@ bool fbgemmSupportedCPU() {
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+// Added for Windows DLL for implicit template parameter instantiation
+template class FBGEMM_API memCopy<std::int32_t, std::int32_t>;
+template class FBGEMM_API DoNothing<std::int32_t, std::int32_t>;
+template class FBGEMM_API DoNothing<float, float>;
+template class FBGEMM_API ReQuantizeForFloat<false, QuantizationGranularity::TENSOR>;
+template class FBGEMM_API ReQuantizeForFloat<false, QuantizationGranularity::GROUP>;
+template class FBGEMM_API ReQuantizeForFloat<false, QuantizationGranularity::OUT_CHANNEL>;
+
+#define INSTANTIATE_BASE(FNAME, RELU, Q_GRAN)    \
+  template class FBGEMM_API FNAME<               \
+    std::uint8_t, std::int32_t,                  \
+    ReQuantizeOutput<RELU, Q_GRAN>>
+
+#define INSTANTIATE_Q_GRAN(FNAME, RELU)                               \
+  INSTANTIATE_BASE(FNAME, RELU, QuantizationGranularity::TENSOR);     \
+  INSTANTIATE_BASE(FNAME, RELU, QuantizationGranularity::GROUP);      \
+  INSTANTIATE_BASE(FNAME, RELU, QuantizationGranularity::OUT_CHANNEL);
+
+#define INSTANTIATE_RELU(FNAME)           \
+  INSTANTIATE_Q_GRAN(FNAME, false);       \
+  INSTANTIATE_Q_GRAN(FNAME, true);
+
+INSTANTIATE_RELU(DoSpmdmOnInpBuffer);
+INSTANTIATE_RELU(DoSConvOnInpBuffer);
+
+#undef INSTANTIATE_RELU
+#undef INSTANTIATE_Q_GRAN
+#undef INSTANTIATE_BASE
+
+#define INSTANTIATE_BASE(RELU, Q_GRAN, BIAS_TYPE)    \
+  template class FBGEMM_API ReQuantizeOutput<RELU, Q_GRAN, BIAS_TYPE>;
+
+#define INSTANTIATE_BIAS_T(RELU, Q_GRAN)       \
+  INSTANTIATE_BASE(RELU, Q_GRAN, std::int32_t) \
+  INSTANTIATE_BASE(RELU, Q_GRAN, float)
+
+#define INSTANTIATE_Q_GRAN(RELU)   \
```

Comment:
Use "clang-format -i *.cc *.h" to apply the clang format here.

Reply:
Done.